### PR TITLE
Replace HTTPretty with Responses

### PR DESCRIPTION
**Pull request by comradekingu**
Attemptedly fixes: #3536
https://github.com/gabrielfalcao/httpretty
https://github.com/getsentry/responses

### CODE REVIEW

File: weblate/accounts/tests/test_registration.py [hunk 1]
```diff
@@ -531,13 +531,13 @@ def test_github(self, confirm=None, fail=False):
             orig_backends = social_django.utils.BACKENDS
             social_django.utils.BACKENDS = GH_BACKENDS
 
-            httpretty.register_uri(
-                httpretty.POST,
+            responses.register_uri(
```
Comment (nijel) [inline weblate/accounts/tests/test_registration.py]:
```suggestion
            responses.add(
```

This method is called differently (and it repeats in basically every test).

File: weblate/machinery/tests.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 import json
 
-import httpretty
+import responses
```
Comment (nijel) [inline weblate/machinery/tests.py]: There are still some httpretty usage left here:

```
   File "/home/runner/work/weblate/weblate/weblate/machinery/tests.py", line 425, in MachineTranslationTest
    @httpretty.activate
NameError: name 'httpretty' is not defined
```

See https://github.com/WeblateOrg/weblate/pull/3542/checks?check_run_id=478710321#step:8:1794

File: weblate/accounts/tests/test_registration.py (hunk 1 above)
Comment (comradekingu) [reply weblate/accounts/tests/test_registration.py]: @nijel Do you know what to do about "has.response"?

Comment (nijel) [reply weblate/accounts/tests/test_registration.py]: No, I didn't yet use much of this library.

File: weblate/machinery/tests.py [hunk 3]
```diff
@@ -409,27 +409,27 @@ def request_callback_post(request, uri, headers):
                 return (200, headers, TERMINOLOGY_LANGUAGES)
             return (200, headers, TERMINOLOGY_TRANSLATE)
 
-        httpretty.register_uri(
-            httpretty.GET,
+        responses.add(
+            responses.GET,
             MST_API_URL,
             body=request_callback_get,
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
            responses.GET,
            MST_API_URL,
            callback=request_callback_get,
```

File: weblate/machinery/tests.py [hunk 4]
```diff
@@ -409,27 +409,27 @@ def request_callback_post(request, uri, headers):
...
             content_type='text/xml',
         )
-        httpretty.register_uri(
-            httpretty.POST,
+        responses.add(
+            responses.POST,
             MST_API_URL,
             body=request_callback_post,
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
            responses.POST,
            MST_API_URL,
            callback=request_callback_post,
```

File: weblate/machinery/tests.py [hunk 5]
```diff
@@ -713,18 +713,18 @@ def test_aws(self):
...
         self.assert_translate(machine, 'es', word='Zkouška')
-        self.assertTrue(httpretty.has_request())
-        httpretty.reset()
+        self.assertTrue(responses.has_request())
+        responses.reset()
         # New instance should use cached languages
         machine = ApertiumAPYTranslation()
         self.assert_translate(machine, 'es')
-        self.assertFalse(httpretty.has_request())
+        self.assertFalse(responses.has_request())
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        self.assertEqual(responses.mock.call_count, 0)
```

File: weblate/machinery/tests.py [hunk 6]
```diff
@@ -713,18 +713,18 @@ def test_aws(self):
...
-    @httpretty.activate
+    @responses.activate
     def test_languages_cache(self):
         machine = self.get_machine(ApertiumAPYTranslation, True)
         self.register_apertium_urls()
         self.assert_translate(machine, 'es')
         self.assert_translate(machine, 'es', word='Zkouška')
-        self.assertTrue(httpretty.has_request())
-        httpretty.reset()
+        self.assertTrue(responses.has_request())
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        self.assertEqual(responses.mock.call_count, 2)
```

File: weblate/machinery/tests.py [hunk 7]
```diff
@@ -659,43 +659,43 @@ def test_saptranslationhub(self):
...
-        httpretty.register_uri(
-            httpretty.POST, 'https://api.deepl.com/v1/translate', body=DEEPL_RESPONSE
+        responses.add(
+            responses.POST, 'https://api.deepl.com/v1/translate', body=DEEPL_RESPONSE
         )
         # Fetch from service
         self.assert_translate(machine, lang='de', word='Hello')
-        self.assertTrue(httpretty.has_request())
-        httpretty.reset()
+        self.assertTrue(responses.has_request())
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        self.assertEqual(responses.mock.call_count, 2)
```

File: weblate/machinery/tests.py [hunk 8]
```diff
@@ -659,43 +659,43 @@ def test_saptranslationhub(self):
...
+        responses.reset()
         # Fetch from cache
         self.assert_translate(machine, lang='de', word='Hello')
-        self.assertFalse(httpretty.has_request())
+        self.assertFalse(responses.has_request())
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        self.assertEqual(responses.mock.call_count, 0)
```

File: weblate/machinery/tests.py [hunk 9]
```diff
@@ -409,27 +409,27 @@ def request_callback_post(request, uri, headers):
                 return (200, headers, TERMINOLOGY_LANGUAGES)
             return (200, headers, TERMINOLOGY_TRANSLATE)
 
-        httpretty.register_uri(
-            httpretty.GET,
+        responses.add(
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        responses.add_callback(
```

File: weblate/machinery/tests.py [hunk 10]
```diff
@@ -409,27 +409,27 @@ def request_callback_post(request, uri, headers):
...
+            responses.GET,
             MST_API_URL,
-            body=request_callback_get,
+            callback=request_callback_get,
             content_type='text/xml',
         )
-        httpretty.register_uri(
-            httpretty.POST,
+        responses.add(
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        responses.add_callback(
```

File: weblate/machinery/tests.py [hunk 11]
```diff
@@ -659,43 +659,43 @@ def test_saptranslationhub(self):
...
-        httpretty.register_uri(
-            httpretty.POST, 'https://api.deepl.com/v1/translate', body=DEEPL_RESPONSE
+        responses.add(
+            responses.POST, 'https://api.deepl.com/v1/translate', body=DEEPL_RESPONSE
         )
         # Fetch from service
         self.assert_translate(machine, lang='de', word='Hello')
-        self.assertTrue(httpretty.has_request())
-        httpretty.reset()
+        self.assertEqual(responses.mock.call_count, 2)
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        self.assertEqual(len(responses.calls), 2)
```

File: weblate/machinery/tests.py [hunk 12]
```diff
@@ -713,18 +713,18 @@ def test_aws(self):
...
-    @httpretty.activate
+    @responses.activate
     def test_languages_cache(self):
         machine = self.get_machine(ApertiumAPYTranslation, True)
         self.register_apertium_urls()
         self.assert_translate(machine, 'es')
         self.assert_translate(machine, 'es', word='Zkouška')
-        self.assertTrue(httpretty.has_request())
-        httpretty.reset()
+        self.assertEqual(responses.mock.call_count, 2)
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        self.assertEqual(len(responses.calls), 2)
```

File: weblate/machinery/tests.py [hunk 13]
```diff
@@ -659,43 +659,43 @@ def test_saptranslationhub(self):
...
         # Fetch from service
         self.assert_translate(machine, lang='de', word='Hello')
-        self.assertTrue(httpretty.has_request())
-        httpretty.reset()
+        self.assertEqual(responses.mock.call_count, 2)
+        responses.reset()
         # Fetch from cache
         self.assert_translate(machine, lang='de', word='Hello')
-        self.assertFalse(httpretty.has_request())
+        self.assertEqual(responses.mock.call_count, 0)
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        self.assertEqual(len(responses.calls), 0)
```

File: weblate/machinery/tests.py [hunk 14]
```diff
@@ -713,18 +713,18 @@ def test_aws(self):
...
         self.assert_translate(machine, 'es', word='Zkouška')
-        self.assertTrue(httpretty.has_request())
-        httpretty.reset()
+        self.assertEqual(responses.mock.call_count, 2)
+        responses.reset()
         # New instance should use cached languages
         machine = ApertiumAPYTranslation()
         self.assert_translate(machine, 'es')
-        self.assertFalse(httpretty.has_request())
+        self.assertEqual(responses.mock.call_count, 0)
```
Comment (nijel) [inline weblate/machinery/tests.py]:
```suggestion
        self.assertEqual(len(responses.calls), 0)
```

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**